src/psyclone/f2pygen.py

```diff
@@ -583,7 +583,7 @@ def __init__(self, parent, content):
         :type parent: :py:class:`psyclone.f2pygen.BaseGen`
         :param str content: the content of the comment
         '''
-        reader = FortranStringReader("! content\n")
+        reader = FortranStringReader("! content\n", ignore_comments=False)
```

Comment:
Is this change required due to you adding support for comments recently and defaulting ignore to True? If so, how did we get away with it until now?

Reply:
I think fparser1 already had support for comments and did so by default. It's only with my recent changes for fparser2 that this has changed - these changes are on master but not in the 0.0.7 release of fparser. This ticket needs the master branch of fparser2 because it needs `Comment` support when adding an OpenACC directive to kernel source.

Reply:
Just to complete what I said above, this means that we will need a new release of fparser before this PR can be merged to master. This will have implications for @MatthewHambley and/or @TeranIvy as the API for fparser2 has changed (I've just been burnt by that in addressing this review).

Reply:
OK, thanks for the info. I think we should change the default to include comments but that is a different issue. Regarding requiring a new release of fparser, we have discussed this and have a new strategy for supporting concurrent development of psyclone and fparser, see pr #213